FWCore/Framework/src/global/OutputModuleBase.cc

```diff
@@ -0,0 +1,380 @@
+      
+      {
+          EventSignalsSentry sentry(act,mcc);
+          write(ep, mcc);
+      }
+      if(remainingEvents_ > 0) {
+        --remainingEvents_;
+      }
+      return true;
+    }
+
+    bool
+    OutputModuleBase::doBeginRun(RunPrincipal const& rp,
+                                 EventSetup const&,
+                                 ModuleCallingContext const* mcc) {
+      doBeginRun_(rp, mcc);
+      return true;
+    }
+    
+    bool
+    OutputModuleBase::doEndRun(RunPrincipal const& rp,
+                               EventSetup const&,
+                               ModuleCallingContext const* mcc) {
+      doEndRun_(rp, mcc);
+      return true;
+    }
+    
+    void
+    OutputModuleBase::doWriteRun(RunPrincipal const& rp,
+                                 ModuleCallingContext const* mcc) {
+      writeRun(rp, mcc);
+    }
+    
+    bool
+    OutputModuleBase::doBeginLuminosityBlock(LuminosityBlockPrincipal const& lbp,
+                                             EventSetup const&,
+                                             ModuleCallingContext const* mcc) {
+      doBeginLuminosityBlock_(lbp, mcc);
+      return true;
+    }
+    
+    bool
+    OutputModuleBase::doEndLuminosityBlock(LuminosityBlockPrincipal const& lbp,
+                                           EventSetup const&,
+                                           ModuleCallingContext const* mcc) {
+      doEndLuminosityBlock_(lbp, mcc);
+      return true;
+    }
+    
+    void OutputModuleBase::doWriteLuminosityBlock(LuminosityBlockPrincipal const& lbp,
+                                                  ModuleCallingContext const* mcc) {
+      writeLuminosityBlock(lbp, mcc);
+    }
+    
+    void OutputModuleBase::doOpenFile(FileBlock const& fb) {
+      openFile(fb);
+    }
+    
+    void OutputModuleBase::doRespondToOpenInputFile(FileBlock const& fb) {
+      doRespondToOpenInputFile_(fb);
+    }
+    
+    void OutputModuleBase::doRespondToCloseInputFile(FileBlock const& fb) {
+      doRespondToCloseInputFile_(fb);
+    }
+    
+    void
+    OutputModuleBase::doPreForkReleaseResources() {
+      preForkReleaseResources();
+    }
+    
+    void
+    OutputModuleBase::doPostForkReacquireResources(unsigned int iChildIndex, unsigned int iNumberOfChildren) {
+      postForkReacquireResources(iChildIndex, iNumberOfChildren);
+    }
+    
+    void
+    OutputModuleBase::preForkReleaseResources() {}
+    
+    void
+    OutputModuleBase::postForkReacquireResources(unsigned int /*iChildIndex*/, unsigned int /*iNumberOfChildren*/) {}
+
+    
+    void OutputModuleBase::maybeOpenFile() {
+      if(!isFileOpen()) reallyOpenFile();
+    }
+    
+    void OutputModuleBase::doCloseFile() {
+      if(isFileOpen()) {
+        reallyCloseFile();
+      }
+    }
+    
+    void OutputModuleBase::reallyCloseFile() {
+    }
+    
+    BranchIDLists const*
+    OutputModuleBase::branchIDLists() const {
```

Comment:
This function really shouldn't be const since it modifies the object held by ```branchIDLists_```. Can you check to see when exactly this method is used?